File: src/main/java/hu/elte/alkfejl/pluto/model/Course.java
package hu.elte.alkfejl.pluto.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
public class Course {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String name;

    @ManyToOne
    @JsonIgnoreProperties(value="courses_T")
    private User teacher;

    @ManyToMany
    @JsonIgnore
    private List<User> students;


    @Transient
    private int studentNumber;

    @ManyToOne
    private Room room;

    private String description;
    private String date;
    public Course(String name,String date,String desc) {
        this.name = name;
        this.date = date;
        this.description = desc;
    }


    public Integer getId() {
        return id;
    }

    public void setTeacher(User teacher) {
        this.teacher = teacher;
    }

    public List<User> getStudents() {
        return students;
    }

    public void setRoom(Room room) {
        this.room = room;
    }

    public User getTeacher() {
        return teacher;
    }

    public void SetNumber() {
        this.studentNumber = students.size();
    }
}

File: client/src/app/model/Course.ts
import {Room} from './Room';
import {User} from './User';
export class Course {
    id: number;
    name: string;
    teacher: User;
    studentNumber: number;
    room: Room;
    date: string;
    description: string;
    constructor(date?: string, teacher?: User, name?: string, room?: Room, startTime?: Date, endTime?: Date, id?: number,studentNumber?: number,description?:string){
        this.room = room;
        this.id = id;
        this.name = name;
        this.teacher = teacher;
        this.date = date;
        this.studentNumber = studentNumber;
        this.description = description;
    }
    
}
